### PR TITLE
skip SW harvest in cap poller when not enabled

### LINKED ISSUE #700: Skip SW harvesting as a result of CAP poller updating when disabled
Do not harvest for SW new authors unless SW is enabled

see https://github.com/sul-dlss/sul_pub/blob/master/lib/cap/authors_poller.rb#L58 to change  analogous to https://github.com/sul-dlss/sul_pub/blob/master/lib/cap/authors_poller.rb#L63

### PR DESCRIPTION
**Pull request by peetucket**
fixes #700

### CODE REVIEW

File: spec/lib/cap/authors_poller_spec.rb [hunk 1]
```diff
@@ -223,4 +223,25 @@
...
+  describe '.do_wos_harvest' do
+    it 'does nothing if WoS client is disabled' do
+      current_wos_setting = Settings.WOS.enabled
+      Settings.WOS.enabled = false
+      subject.do_wos_harvest
+      expect(Author).not_to receive(:where)
+      expect(WebOfScience.harvester).not_to receive(:harvest)
+      Settings.WOS.enabled = current_wos_setting
+    end
+  end
```
Comment (darrenleeweber) [inline spec/lib/cap/authors_poller_spec.rb]: :+1: to adding specs for this :balloon: 

Consider using `allow(Settings.WOS).to receive(:enabled).and_return(true)` (or false) to mock setting these values.

Comment (darrenleeweber) [reply spec/lib/cap/authors_poller_spec.rb]: It might simplify to `expect(WebOfScience).not_to receive(:harvester)`

Comment (darrenleeweber) [reply spec/lib/cap/authors_poller_spec.rb]: I usually put an `expect` before calling something to ensure it's mocked, e.g.
```
allow(Settings.WOS).to receive(:enabled).and_return(false)
expect(WebOfScience).not_to receive(:harvester)
subject.do_wos_harvest
```

Comment (peetucket) [reply spec/lib/cap/authors_poller_spec.rb]: thanks for the suggestions, the mocking of settings is a much better way of doing that... all made, rebased and pushed

**Pull request merged**